src/lib.rs

```diff
@@ -679,8 +679,8 @@ impl<T: AsFd> Async<T> {
         let registration = unsafe { Registration::new(fd) };
 
         Ok(Async {
-            source: Reactor::get().insert_io(registration)?,
-            io: Some(io),
+            source: SourceContainer(Reactor::get().insert_io(registration)?),
```

Comment:
Rather than doing this, I'd prefer for `SourceContainer` to implement `From<Arc<Source>>`. That way it can be constructed with `.into()` instead of manually like this.

Reply:
Done.